### PR TITLE
Stats: Remove the feature flag for the restricted dashboard

### PR DESCRIPTION
**Pull request by dognose24**
Related to https://github.com/Automattic/red-team/issues/26

## Proposed Changes

* Remove the feature flag `stats/restricted-dashboard` and its usage.
* Determine if the paywall force redirection should apply to pages wrapped with `LoadStatsPage` other than Traffic and Insights.

## Why are these changes being made?

* Remove the outdated feature flag as a maintenance.

## Testing Instructions

* Spin this change up with the Calypso Live branch.
* Ensure the Stats modules and cards for the restricted dashboard work as previously.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/features/modules/stats-devices/stats-module-devices-wrapper.tsx [hunk 1]
```diff
@@ -22,19 +22,17 @@ const StatsModuleDevicesWrapper: React.FC< StatsAdvancedModuleWrapperProps > = (
...
 	const { isPending: isFetchingUsage, data: usageData } = usePlanUsageQuery( siteId );
-	const { isLoading: isLoadingFeatureCheck, supportCommercialUse } = useStatsPurchases( siteId );
+	const { isLoading: isLoadingFeatureCheck } = useStatsPurchases( siteId );
 
 	const isSiteInternal = ! isFetchingUsage && usageData?.is_internal;
 	const isFetching = isFetchingUsage || isLoadingFeatureCheck; // This is not fetching Devices data.
 
-	const isAdvancedFeatureEnabled =
-		isSiteInternal || ( isGatedByShouldGateStats ? ! shouldGateStats : supportCommercialUse );
+	const isAdvancedFeatureEnabled = isSiteInternal || ! shouldGateStats;
```
Comment (grzegorz-cp) [inline client/my-sites/stats/features/modules/stats-devices/stats-module-devices-wrapper.tsx:35]: Won't this expose the advanced features to everyone?

Comment (dognose24) [reply client/my-sites/stats/features/modules/stats-devices/stats-module-devices-wrapper.tsx:35]: No, the Devices module is gated with [shouldGateStats](https://github.com/Automattic/wp-calypso/pull/93384/files/c2af4de6788c32f007e38b189088b073ca54f2d6#diff-0e11044d19ddadbb5f285305ca03b520e15442312488397144df4a2cf7e61860R26) here.

File: client/my-sites/stats/stats-redirect/index.tsx [hunk 2]
```diff
@@ -73,8 +70,9 @@ const StatsRedirectFlow: React.FC< StatsRedirectFlowProps > = ( { children } ) =
 	}, [ dispatch, siteId, isLoadingNotices, purchaseNotPostponed ] );
 
 	// Render conditions (for readability).
+	// TODO: Determine the paywall redirection for specific routes.
 	const shouldRenderPaywall =
-		! isLoading && ! skipPaywallFlow && redirectToPurchase && siteSlug && canUserManageOptions;
+		false && ! isLoading && redirectToPurchase && siteSlug && canUserManageOptions;
```
Comment (kangzj) [inline client/my-sites/stats/stats-redirect/index.tsx:75]: It seems the summary pages are all properly gated:

image


Let's revisit whether there are some pages that still need the paywall. If not, we'll be able to get rid of the legacy code to optimize bundle size.

**Review by kangzj — APPROVED** [review]
LGTM!

File: client/my-sites/stats/stats-redirect/index.tsx (hunk 2 above)
Comment (kangzj) [reply client/my-sites/stats/stats-redirect/index.tsx:75]: Tracked [here](https://github.com/Automattic/red-team/issues/139).

**Pull request merged**